notebooks/knn.ipynb

```diff
@@ -0,0 +1,447 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "db673eda-86ee-47a7-975d-aa3d36c2f407",
+   "metadata": {},
+   "source": [
+    "# KNN"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "eeb78fae-ae08-4b64-8daa-b579af1d9ba6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "\n",
+    "from spark_rapids_ml.knn import NearestNeighbors\n",
+    "from pyspark.sql.functions import col"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17955bd7-b911-4da6-ad1b-02ae4a5f0a2b",
+   "metadata": {},
+   "source": [
+    "### Create synthetic dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "38c671ba-b6de-4414-992f-957e28f3a8be",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "dim = 2000\n",
```

Comment:
Does this work on Databricks?

Reply:
Just tried.  Yes, although it looks like there might be a bug w/ `exactNearestNeighborsJoin` returning no results (even at smaller data sizes).